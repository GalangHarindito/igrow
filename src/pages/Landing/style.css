.landing {
  padding: 1rem;
  
}

.landing > h3 {
  text-align: center;
  padding: 2rem 0;

}

.landing > section {
  display: flex;
  justify-content: center;
  margin: 2rem 0;
}

@media screen and (max-width: 700px) {
  .landing > section {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  
}