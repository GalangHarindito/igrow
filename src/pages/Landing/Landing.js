import React, { useState } from "react";
import Card from "../../component/Card";
import axios from "axios";
import "./style.css";
import Button from "../../component/Button";
import LoadingCard from "../../component/LoadingCard";

export default function Landing() {
  const [state, setState] = useState([]);
  const [loading, setLoading] = useState(false);

  const fetchApi = () => {
    setLoading(true);
    const options = {
      method: "GET",
      url: `https://randomuser.me/api/?results=2`,
      headers: {},
    };

    axios(options)
      .then((res) => {
        setLoading(false);
        setState(res.data.results);
      })
      .catch((err) => {
        //if(status === 401){
        //  window.location.href = '/login'
        //}
        //const messageStatus = status > 401 && status <= 500 ? 'Sedang ada masalah, silahkan refresh halaman' : message;
      });
  };

  return (
    <section className='landing'>
      <h3>Find Your Friends</h3>
      {loading ? <LoadingCard data={state} /> : <Card data={state} />}
      <section>
        <Button
          className='find'
          label="Let's Find"
          onClick={() => fetchApi()}
        />
      </section>
    </section>
  );
}
