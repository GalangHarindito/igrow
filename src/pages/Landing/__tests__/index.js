import React, { useState } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Landing, { fetchApi } from '../Landing';

describe('src/pages/Landing', () => {

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Landing />);
    expect(tree).toMatchSnapshot();
  });

  //it('fetch', async () => {
  //  const data = [
  //    {
  //      name : 'test',
  //      gender: 'test',
  //      phone: 'test',
  //      picture: { large:'test' },
  //      location: { city: 'test', country: 'test'}
  //    },
  //    {
  //      name : 'test',
  //      gender: 'test',
  //      phone: 'test',
  //      picture: { large:'test' },
  //      location: { city: 'test', country: 'test'}
  //    }
  //  ];
 
  //  axios.get.mockImplementationOnce(() => Promise.resolve(data));
 
  //  await expect(fetchApi()).toEqual(data);
 
  //  expect(axios.get).toHaveBeenCalledWith(
  //    `https://randomuser.me/api/?results=2`,
  //  );
  //});


});
