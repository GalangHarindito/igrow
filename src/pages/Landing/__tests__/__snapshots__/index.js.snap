// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/Landing render 1`] = `
<section
  className="landing"
>
  <h3>
    Find Your Friends
  </h3>
  <Card
    data={Array []}
  />
  <section>
    <Button
      className="find"
      isLoading={false}
      label="Let's Find"
      onClick={[Function]}
    />
  </section>
</section>
`;
